python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py

```diff
@@ -51,24 +52,24 @@ def forward(self, x):
         print(sys.argv)
         print("Need parameters: <imagePath>")
         exit(-1)
-
-    hadoop_conf_dir = os.environ.get('HADOOP_CONF_DIR')
-
-    if hadoop_conf_dir:
-        num_executors = 2
-        num_cores_per_executor = 4
-        zoo_conda_name = detect_conda_env_name()  # auto detect current conda env name
-        sc = init_spark_on_yarn(
-            hadoop_conf=hadoop_conf_dir,
-            conda_name=zoo_conda_name,
-            num_executors=num_executors,
-            executor_cores=num_cores_per_executor,
-            executor_memory="8g",
-            driver_memory="2g",
-            driver_cores=1)
-    else:
-        num_cores_per_executor = 4
-        sc = init_spark_on_local(cores=num_cores_per_executor, conf={"spark.driver.memory": "10g"})
+    parser = OptionParser()
+    parser.add_option("--executor-cores", type=int, dest="cores", default=4, help="number of executor cores")
+    parser.add_option("--num-executors", type=int, dest="executors", default=16, help="number of executors")
+    parser.add_option("--executor-memory", type=str, dest="executorMemory", default="30g", help="executor memory")
+    parser.add_option("--driver-memory", type=str, dest="driverMemory", default="30g", help="driver memory")
+    parser.add_option("--deploy-mode", type=str, dest="deployMode", default="yarn-client", help="yarn deploy mode, yarn-client or yarn-cluster")
+    (options, args) = parser.parse_args(sys.argv)
+
+    hadoop_conf = os.environ.get('HADOOP_CONF_DIR')
+    assert hadoop_conf, "Directory path to hadoop conf not found for yarn-client mode. Please " \
+            "set the environment variable HADOOP_CONF_DIR"
+
+    conf = create_spark_conf().set("spark.executor.memory", options.executorMemory)\
```

Comment:
use init_orca_context's build-in parameter

Reply:
OK, I will update.